transcom/mymove
# INT B-20591 prime can update destination address for NTSR shipment

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20591)

## Summary

The gov is wanting NTSR shipments to go through the same process as HHG shipments when the Prime requests a change to the delivery/destination address with the end goal being the TOO needs to approve the change if the business rules change.

This PR does the following:
- some random white space deletions cause it was driving me nuts
- added on to existing query when looking for shipments that are requesting an addy change - for NTSR shipments we need the storage address since technically that's the "pickup" address
- adjusted logic checks to either look at pickup or storage address, depending on `shipmentType`
- updated UI to conditionally show `Review required` instead of the address for NTSR shipments that have a pending destination address change
- added a way for the office user to click the `See delivery location to proceed.` link and be scrolled to where they need their attention
- disabled destination fields when the NTSR shipment has a pending destination address request
- let the TOO open up the destination address request modal for NTSR shipments
- minor CSS changes
- updated/added relevant tests to both backend and UI

### How to test

1. Fire up yer `client` & `server`
2. Find any shipment as a TOO (my fav is `SITEXT`) and add on a NTSR shipment
3. Fill out all of the necessary information
4. Make sure you approve the shipment (this makes it available to the Prime)
5. Go into the Prime Simulator, find the shipment inside the move, and click the `Update Shipment Destination Address` button
6. Fill out the form with an address that's hecka far away from the current ZIP, save it
7. Go into the same move as the TOO and you should see the shipment have a `Review required` where the NTSR's destination address should be
8. Go through the flow and either approve/reject the shipment (approving changes the address)
9. Now go through steps again and change the ZIP to be somewhere or the same ZIP - you shouldn't need approval from the TOO now since the business rules don't change


## Screenshots
![Screenshot 2024-07-29 at 11 08 58 AM](https://github.com/user-attachments/assets/1c3869a3-fe9b-4687-9471-53c33977f84a)

https://github.com/user-attachments/assets/42470d73-6ff0-4933-8bc9-0880e1813bf3
